### DISCUSSION TITLE
Non-bold title & headings in Typst PDFs.

### DISCUSSION THREAD

**Discussion by the-solipsist** [quarto-dev/quarto-cli#8859] [Q&A]
### Description

I'm trying out Typst as the PDF engine instead of xelatex / lualatex.  With LaTeX, I generally use `documentclass: article`, which uses non-bold document title and section headings.  I want to replicate that with Typst.  

I looked at Quarto's Typst template and saw that it wasn't adding any bold styling to the titles / section headings, so it seems Typst is doing that on its own.
https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/formats/typst/pandoc/typst.template

But I'm sure there _must_ be a way to remove bold styling from the titles / section headings. Would anyone know how?

**Comment by the-solipsist** [comment]
To answer my own question, I was looking at the typst.template, whilst I should have been looking at template.typst

https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/formats/typst/pandoc/template.typst
and
https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/formats/typst/pandoc/quarto/typst-template.typ

Both of those have:

```
  if title != none {
    align(center)[#block(inset: 2em)[
      #text(weight: "bold", size: 1.5em)[#title]
    ]]
  }
```

But even there, I don't see where the bold styling is being applied to the section headings.

```
  set heading(numbering: sectionnumbering)
```

I'll try modifying the template partial to remove the `weight: "bold"` and report back.

**Comment by the-solipsist** [comment]
I'm still having difficulty getting this working. So any help would be appreciated.

**Reply by mcanouil** [reply]
Hello, thank you for your interest in Quarto. We appreciate your feedback regarding `typst` format. However, we would like to kindly remind you that we are not available 24/7, and we also need some time off during the weekends. Therefore, we ask you to be patient and respectful of our working hours and days. We will try our best to respond to your request as soon as possible, but we cannot guarantee a quick turnaround. Thank you for your understanding and cooperation.

**Reply by the-solipsist** [reply]
Hi, @mcanouil. I'm sorry, I didn't mean to sound impatient, and I don't even have an expectation that all questions will be answered. After all, people are volunteering their time on this discussion board.

In my previous message, I had promised to report back after my experimentation... and I was just reporting back that I hadn't been successful, and so could do with others' help.

**Comment by mcanouil** [comment]
Did you read <https://typst.app/docs/tutorial/advanced-styling/>?
Quarto(Pandoc) is only borrowing the code/language as it does for LaTeX/HTML/OpenXML. So understanding the language itself and reading its documentation is required when advanced styling of a document is wanted.
There are no workarounds I'm afraid.
For instance, in `typst-show.typ`[^1]:

```typst
#show heading: it => block(width: 100%)[
  #set text(weight: "regular")
  #(it.body)
]
```

<table>
<tr><th>Quarto document</th><th>Typst/PDF</th></tr>
<tr><td>

````qmd
---
title: "A title"
format: typst
template-partials:
  - typst-show.typ
---

## Introduction

lorum ipsum dolor sit amet

### Subsection

lorum ipsum dolor sit amet

## Methods

lorum ipsum dolor sit amet

## Results

lorum ipsum dolor sit amet

## Discussion

lorum ipsum dolor sit amet

## Conclusion

lorum ipsum dolor sit amet

````

</td><td>

<img width="293" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/5b434f17-2ccb-4d89-94ba-b9ef51e547ef">

</td></tr>
</table>

Or even without partials:

<table>
<tr><th>Quarto document</th><th>Typst/PDF</th></tr>
<tr><td>

````qmd
---
title: "A title"
format: typst
include-before-body:
  text: |
    #show heading: it => block(width: 100%)[
      #set text(weight: "regular")
      #(it.body)
    ]
---

## Introduction

lorum ipsum dolor sit amet

### Subsection

lorum ipsum dolor sit amet

## Methods

lorum ipsum dolor sit amet

## Results

lorum ipsum dolor sit amet

## Discussion

lorum ipsum dolor sit amet

## Conclusion

lorum ipsum dolor sit amet

````

</td><td>

<img width="293" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/74ed1dea-4785-4bb8-b4c1-4988992c4728">

</td></tr>
</table>

[^1]: code is directly derived from the Typst documentation.

**Comment by the-solipsist** [comment]
Thanks. 

Yes, I had read https://typst.app/docs/tutorial/advanced-styling/ and https://quarto.org/docs/output-formats/typst.html#includes . 

I got the headings to be non-bold by using a slightly simpler syntax:

```
    include-before-body:
      text: |
        #show heading: set text(weight: "regular")
```

But I still couldn't figure out how to style the title, since there is no "title" equivalent to "heading" in Typst.  E.g., this (within `include-before-body`) won't work: `#show text(weight: "regular")[#title]`. I tried to initialize "title", by adding `#let title = [""] \n #show title: set text(weight: "regular")`, but that didn't work either.

Any further suggestions are welcome. Thanks.

**Reply by mcanouil** [reply]
You already know:

https://github.com/quarto-dev/quarto-cli/blob/b222df50198848437ebcaa2b6489b4b29ddfc907/src/resources/formats/typst/pandoc/quarto/typst-template.typ#L33-L37

So you know already you have to change the partials/template.
What else do you need exactly?

<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/73a62669-7b3b-48fd-b080-3aa2c7194a92">
